### DISCUSSION TITLE
Semantic search for ChatGPT embedding based answers. AI answers based on specific outline documentation!

### DISCUSSION THREAD

**Discussion by kb1900** [outline/outline#5086] [Ideas]
Title kind of explains overall idea. Host workspace knowledge on outline as a wiki, use this wiki to provide embedding contexts (paragraphs, sentences, full documents) to chat ChatGPT AI for highly trustable answers to questions with referenced links/documents. 

OpenAI cookbook has a fairly detailed explanation of how the implementation could work and how useful this could be: https://github.com/openai/openai-cookbook/blob/main/examples/Question_answering_using_embeddings.ipynb

I am a complete hobbyist programmer but think this would be a fantastic low hanging fruit to target as a killer feature! Curious what others think or if anyone has implemented anything similar.

Notably semantic search is not necessarily better than keyword search for this but made more sense in my mind for my usecase. I see that search already has an easily accessible endpoint via the api. When I have some time I may spin up a proof of concept using the cookbook link above as a guide with the search api endpoint for contexts/embeddings

**Comment by tommoor** [comment]
We're working on this 😉

**Comment by chenyf** [comment]
非常期待

**Comment by mmlkrx** [comment]
@tommoor

I'm curious how this is coming along and if you are hoping to support other LLMs besides OpenAI models as well?

**Reply by tommoor** [reply]
I expect the feature to be opt-in in cloud-hosted Outline in the next few weeks, for on-premise it's a bit trickier but it will definitely be available in the next 2 releases.

**Reply by Nathan13888** [reply]
Any ideas where's the code for this upcoming feature, and if there's a job board listing some things others could (potentially) contribute?

**Reply by superkelvint** [reply]
@tommoor By on-premise, do you mean the open-source version available on github?

**Reply by tommoor** [reply]
This feature will only be included in paid editions, I'm not planning on putting it in the community edition

**Reply by superkelvint** [reply]
Understood, thanks.
